core/trino-main/src/test/java/io/trino/execution/TestAddColumnTask.java

```diff
@@ -156,18 +164,130 @@ public void testAddColumnOnMaterializedView()
         QualifiedObjectName materializedViewName = qualifiedObjectName("existing_materialized_view");
         metadata.createMaterializedView(testSession, QualifiedObjectName.valueOf(materializedViewName.toString()), someMaterializedView(), false, false);
 
-        assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(materializedViewName), new Identifier("test"), INTEGER, Optional.empty(), false, false)))
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(materializedViewName), QualifiedName.of("test"), INTEGER, Optional.empty(), false, false)))
                 .hasErrorCode(TABLE_NOT_FOUND)
                 .hasMessageContaining("Table '%s' does not exist", materializedViewName);
     }
 
-    private ListenableFuture<Void> executeAddColumn(QualifiedName table, Identifier column, Type type, Optional<String> comment, boolean tableExists, boolean columnNotExists)
+    @Test
+    public void testAddFieldWithNotExists()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, rowTable(tableName, new RowType.Field(Optional.of("a"), BIGINT)), false);
+        TableHandle table = metadata.getTableHandle(testSession, tableName).get();
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .containsExactly(new ColumnMetadata("col", rowType(new RowType.Field(Optional.of("a"), BIGINT))));
+
+        getFutureValue(executeAddColumn(asQualifiedName(tableName), QualifiedName.of("col", "a"), INTEGER, false, true));
+        assertThat(metadata.getTableMetadata(testSession, table).getColumns())
+                .containsExactly(new ColumnMetadata("col", rowType(new RowType.Field(Optional.of("a"), BIGINT))));
+    }
+
+    @Test
+    public void testAddFieldToNotExistingField()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        metadata.createTable(
+                testSession,
+                TEST_CATALOG_NAME,
+                rowTable(tableName, new RowType.Field(Optional.of("a"), rowType(new RowType.Field(Optional.of("b"), INTEGER)))),
+                false);
+
+        assertTrinoExceptionThrownBy(() -> getFutureValue(executeAddColumn(asQualifiedName(tableName), QualifiedName.of("col", "x", "c"), INTEGER, false, false)))
+                .hasErrorCode(NOT_SUPPORTED)
+                .hasMessage("Cannot resolve field 'x' within row(a row(b integer)) type when adding [x, c] in row(a row(b integer))");
```

Comment:
The added field should render like a dereference expression, not like a list:
`when adding [x, c]` -> `when adding x.c`